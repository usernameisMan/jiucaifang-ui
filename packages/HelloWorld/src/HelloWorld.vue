<template>
  <div>
    <p class="demo-box">
      This is {{ t("vgc.helloworld.hello") }} {{ message }}
    </p>
  </div>
</template>

<script>
import Locale from "../../../src/utils/mixins/locale.js";
export default {
  name: "HelloWorld",
  mixins: [Locale],
  props: {
    message: {
      type: String,
      default: "world",
    },
  },
};
</script>

<style lang="scss" scoped>
.demo-box {
  font-size: 24px;
  font-weight: bold;
}
</style>
