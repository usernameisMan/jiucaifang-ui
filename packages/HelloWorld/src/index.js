import HelloWorld from "./HelloWorld.vue"

HelloWorld.install = (Vue) => {
  Vue.component(HelloWorld.name, HelloWorld)
}

export default HelloWorld