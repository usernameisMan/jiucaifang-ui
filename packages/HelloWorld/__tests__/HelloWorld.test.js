
import { mount } from '@vue/test-utils'
import HelloWorld from "../src/HelloWorld";

describe("HelloWorld.test.js", () => {
  // 现在挂载组件，你便得到了这个包裹器
  const wrapper = mount(HelloWorld, {
    propsData: {
      message: 'World1'
    }
  })

  it('message includes "world"', () => {
    expect(wrapper.html()).toContain("World1");
  });

  it("has the expected html structure", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
