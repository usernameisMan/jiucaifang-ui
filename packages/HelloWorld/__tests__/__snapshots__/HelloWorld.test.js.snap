// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.test.js has the expected html structure 1`] = `
<div>
    <p class="demo-box">
        This is 你好 World1
    </p>
</div>
`;
